infoschema/builder.go

```diff
@@ -40,13 +42,16 @@ type Builder struct {
 // Return the detail updated table IDs that are produced from SchemaDiff and an error.
 func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, error) {
 	b.is.schemaMetaVersion = diff.Version
-	if diff.Type == model.ActionCreateSchema {
+	switch diff.Type {
+	case model.ActionCreateSchema:
 		return nil, b.applyCreateSchema(m, diff)
-	} else if diff.Type == model.ActionDropSchema {
+	case model.ActionDropSchema:
 		tblIDs := b.applyDropSchema(diff.SchemaID)
 		return tblIDs, nil
-	} else if diff.Type == model.ActionModifySchemaCharsetAndCollate {
+	case model.ActionModifySchemaCharsetAndCollate:
 		return nil, b.applyModifySchemaCharsetAndCollate(m, diff)
+	case model.ActionAlterTableAlterPartition:
```

Comment:
Do we need to this when the action is `ActionDropTablePartition` or `ActionTruncateTablePartition`?

Reply:
That's the 3rd and 4rd PR.

Reply:
Got it.